libs/openFrameworks/types/ofBaseTypes.h

```diff
@@ -621,218 +621,1051 @@ class ofBaseVideoPlayer: virtual public ofBaseVideo{
+	/// call to setColor() changes the drawing color.
+	///
+	/// \param color The color to use when drawing.
+	/// \param _a The alpha value between 0 and 255 to use when drawing.
 	virtual void setColor(const ofColor & color, int _a)=0;
-	virtual void setColor(int gray)=0; // new set a color as grayscale with one argument
-	virtual void setHexColor( int hexColor )=0; // hex, like web 0xFF0033;
+	/// \brief Set the global color this renderer will use when drawing.
+	///
+	/// The renderer will continue using a color set by setColor() until another
+	/// call to setColor() changes the drawing color.
+	///
+	/// \param gray The grayscale value to use when drawing.
+	virtual void setColor(int gray)=0;
+	/// \brief Set the global color this renderer will use when drawing.
+	///
+	/// The renderer will continue using a color set by setColor() until another
+	/// call to setColor() changes the drawing color.
+	///
+	/// \param hexColor The hexidecimal representation of the color to use when
+	/// drawing.
+	virtual void setHexColor( int hexColor )=0;
 
+	/// \brief Set this renderer's bitmap text mode.
+	///
+	/// Possible bitmap texture modes include:
+	///		OF_BITMAPMODE_SIMPLE
+	///		OF_BITMAPMODE_SCREEN
+	///		OF_BITMAPMODE_VIEWPORT
+	///		OF_BITMAPMODE_MODEL
+	///		OF_BITMAPMODE_MODEL_BILLBOARD
+	///
+	/// \param mode The bitmap mode to request this renderer to use.
+	/// \sa ofDrawBitmapMode
 	virtual void setBitmapTextMode(ofDrawBitmapMode mode)=0;
 
-	// bg color
+	/// \brief Get this renderer's current background color.
+	/// \returns This renderer's current background color.
 	virtual ofColor getBackgroundColor()=0;
+	/// \brief Set this renderer's background color.
+	/// \param c The color to request this renderer to use.
 	virtual void setBackgroundColor(const ofColor & c)=0;
+	
+	/// \brief Immediately paint a background color to the screen.
+	///
+	/// If automatic background drawing is enabled (which it is by default) this
+	/// method called from ofApp::setup() will also repaint the background with
+	/// this color each frame.
+	///
+	/// \param c The color to paint the background with.
 	virtual void background(const ofColor & c)=0;
+	/// \brief Immediately paint a grayscale background color to the screen.
+	///
+	/// If automatic background drawing is enabled (which it is by default) this
+	/// method called from ofApp::setup() will also repaint the background with
+	/// this color each frame.
+	///
+	/// \param brightness The grayscale value between 0 and 255 to paint the
+	/// background with.
 	virtual void background(float brightness)=0;
+	/// \brief Immediately paint a grayscale background color to the screen.
+	///
+	/// If automatic background drawing is enabled (which it is by default) this
+	/// method called from ofApp::setup() will also repaint the background with
+	/// this color each frame.
+	///
+	/// \param hexColor The 24-bit hex representation of the color to paint the
```

Comment:
Is this an appropriate way to describe the `int` representation of 24-bit color, or is this confusing because `ints` on most systems are actually 32- and 64-bit?